utils.go

```diff
@@ -0,0 +1,15 @@
+package tests
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io"
+)
+
+func calculateHashFromReader(r io.Reader) string {
```

Comment:
How about comparing md5 bytes directly instead of calculating the hex? We can use `ShouldResemble` here.